pkg/simple_cache/lib/src/providers/redis.dart

```diff
@@ -0,0 +1,112 @@
+import 'dart:async';
+import 'dart:io' show SocketException;
+import 'package:dartis/dartis.dart';
+import 'package:retry/retry.dart';
+import '../../cache_provider.dart';
+
+/// Create a [CloseableCacheProvider] with redis as backing store.
+CloseableCacheProvider<List<int>> redisCacheProvider(
+  String connectionString, {
+  RetryOptions retryOptions = const RetryOptions(),
+  Duration connectTimeLimit = const Duration(seconds: 30),
+  Duration commandTimeLimit = const Duration(seconds: 5),
+}) {
+  final provider = _RedisCacheProvider(
+    connectionString,
+    retryOptions: retryOptions,
+    connectTimeLimit: connectTimeLimit,
+    commandTimeLimit: commandTimeLimit,
+  );
+  return CloseableCacheProvider.withClose(provider, provider.disconnect);
+}
+
+class _RedisContext {
+  final Connection connection;
+  final Client client;
+  final Commands<String, List<int>> commands;
+  _RedisContext({this.connection, this.client, this.commands});
+}
+
+typedef _op<T> = Future<T> Function(Commands<String, List<int>>);
+
+class _RedisCacheProvider extends CacheProvider<List<int>> {
+  final String _connectionString;
+  final RetryOptions _r;
+  final Duration _connectTimeLimit;
+  final Duration _commandTimeLimit;
+
+  _RedisCacheProvider(
+    this._connectionString, {
+    RetryOptions retryOptions = const RetryOptions(),
+    Duration connectTimeLimit = const Duration(seconds: 30),
+    Duration commandTimeLimit = const Duration(seconds: 5),
+  })  : _r = retryOptions,
+        _connectTimeLimit = connectTimeLimit,
+        _commandTimeLimit = commandTimeLimit {
+    assert(_r != null, 'RetryOptions must be given');
+    assert(_connectionString.isNotEmpty, 'connectionString must be given');
+    assert(!_connectTimeLimit.isNegative, 'connectTimeLimit is negative');
+    assert(!_commandTimeLimit.isNegative, 'commandTimeLimit is negative');
+  }
+
+  Future<_RedisContext> _ctx;
+
+  Future<_RedisContext> _connect() => _r.retry(() async {
+        // Attempt to make a connection, timeout after _connectTimeLimit
+        final conn = await Connection.connect(_connectionString)
+            .timeout(_connectTimeLimit);
+        // When connection is closed, error or somehow we reset _ctx.
+        // Any errors have already been returned if any commands was affect
+        // otherwise the error is just a broken TCP connection and we ignore it.
+        conn.done
```

Comment:
`conn.done.whenComplete()`? it looks like you won't be using neither the completion value nor the error.